source/delegate/contracts/Delegate.sol

```diff
@@ -167,6 +169,14 @@ contract Delegate is IDelegate, Ownable {
       amountToStake,
       bytes32(uint256(address(this)) << 96) //NOTE: this will pad 0's to the right
     );
+
+    if (oldStakingAmount > amountToStake) {
```

Comment:
should we place this block above the external `indexer.setIntent()` call?

Reply:
we can't because the staked amount from the indexer would not be in this contract yet.

Reply:
of course, thanks for the clarification.